enterprise/management/commands/email_drip_for_missing_dsc_records.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+"""
+Django management command for sending an email to learners with missing DataSharingConsent records.
+"""
+import logging
+from datetime import timedelta
+
+from django.core.management import BaseCommand
+from django.utils import timezone
+
+from consent.models import DataSharingConsent, ProxyDataSharingConsent
+from enterprise import utils
+from enterprise.models import EnterpriseCourseEnrollment
+
+LOGGER = logging.getLogger(__name__)
+
+PAST_NUM_DAYS = 1
+
+
+class Command(BaseCommand):
+    """
+    Django management command for sending an email to learners with missing DataSharingConsent records
+    """
+
+    def handle(self, *args, **options):
+        current_minus_two_days = timezone.now().date() - timedelta(days=PAST_NUM_DAYS)
+
+        enterprise_course_enrollments = EnterpriseCourseEnrollment.objects.filter(
+            created__gte=current_minus_two_days
+        )
+        for enterprise_enrollment in enterprise_course_enrollments:
+            username = enterprise_enrollment.enterprise_customer_user.username
+            user_id = enterprise_enrollment.enterprise_customer_user.user_id
+            course_id = enterprise_enrollment.course_id
+            enterprise_customer = enterprise_enrollment.enterprise_customer_user.enterprise_customer
+            consent = DataSharingConsent.objects.proxied_get(
+                username=username,
+                course_id=course_id,
+                enterprise_customer=enterprise_customer
+            )
+            if isinstance(consent, ProxyDataSharingConsent):
+                utils.track_event(user_id, 'edx.bi.user.consent.absent', {
+                    'course': course_id,
+                    'username': username,
+                    'enterprise_name': enterprise_customer.name,
+                    'enterprise_uuid': enterprise_customer.uuid
+                })
+                LOGGER.info('Email sent for missing data sharing consent. User: [%s], Course: [%s], Enterprise: [%s]',
```

Comment:
We are following a ritual of adding a prefix for logging. It will make our debugging easy in the future. For reference, you can see the [NudgeEmail management command](ecommerce/enterprise/management/commands/send_code_assignment_nudge_emails.py).

Reply:
Done